src/test/java/DataDriven/AutoFillForm.java

```diff
@@ -0,0 +1,226 @@
+package DataDriven;
+
+import org.testng.annotations.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvValidationException;
+
+import DataDriven.framework.FormModelObject;
+import TheInternet.foundation.ConfigurationReader;
+import TheInternet.foundation.TestBase;
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+
+public class AutoFillForm extends TestBase{
```

Comment:
Notice that all of the tests contain the same steps. This is a feature of Data-Driven testing - same steps, different data source.
I would move the steps to common method in the TestClass.